ERCS/erc-xxxx.md

```diff
@@ -0,0 +1,201 @@
+
+1. With the `atomicBatch` capability enabled
+2. With the `chainId` set to the same value in all calls
+
+### Static Paymaster Configuration Capability
+
+Note that use of Paymasters managed by a "paymaster web service" is described in [ERC-7677](./eip-7677).
+
+Identifier:
+
+`staticPaymasterConfiguration`
+
+Interface:
+
+```typescript
+type StaticPaymasterConfigurationCapabilityParams = Record<
+  `0x${string}`, // Chain ID
+  {
+    paymaster: string;
+    paymasterData: string;
+    paymasterValidationGasLimit: `0x${string}`;
+    paymasterPostOpGasLimit: `0x${string}`;
+  }
+>;
+```
+
+### On-chain Query Paymaster Configuration Capability
+
+Identifier:
+
+`onChainQueryPaymasterConfiguration`
+
+Interface:
+
+```typescript
+type OnChainQueryPaymasterConfigurationCapabilityParams = Record<
+  `0x${string}`, // Chain ID
+  {
+    target: `0x${string}`, // contract to query for Paymaster configuration
+    context: `0x${string}`, // hex-encoded byte array to pass to the configuration provider
+  }
+>
+```
+
+We then define the following Solidity interface:
+
+```solidity
+
+    struct PaymasterConfiguration {
+        address paymaster;
+        bytes paymasterData;
+        uint256 paymasterValidationGasLimit;
+        uint256 paymasterPostOpGasLimit;
+    }
+
+interface IPaymasterConfigurationResolver {
+    function getPaymasterConfiguration(bytes operation, bytes context) external;
+}
+
+```
+
+The wallet MUST perform an ABI-encoding of the entire `operation` object and make a view call
+to the `getPaymasterConfiguration` function of the `target` address.
+
+If the view call fails or returns an invalid data, the `wallet_sendCalls` method must fail and return with an error.
+
+### Validity Time Range Capability
+
+Identifier:
+
+`validityTimeRange`
+
+Interface:
+
+```typescript
+type ValidityTimeRangeCapabilityParams = Record<
+  `0x${string}`, // Chain ID
+  {
+    validAfter: `0x${string}`, // operation valid only after this timestamp, in seconds
+    validUntil: `0x${string}`  // operation valid only before this timestamp, in seconds
+  }
+>
+```
+
+The wallet must then verify the time range [`validAfter`..`validUntil`] is valid and present it to the
+user in a human-readable way for confirmation as part of the transaction information.
+
+### Multidimensional Nonce Capability
+
+Identifier:
+
+`multiDimensionalNonce`
+
+Interface:
+
+```typescript
+type MultiDimensionalNonceCapabilityParams = Record<
+  `0x${string}`, // Chain ID
+  {
+    nonceKey: `0x${string}`,
+    nonceSequence: `0x${string}`
+  }
+>
+```
+
+For Smart Contract Accounts that support multidimensional nonce values,
+the wallet must specify these parameters during the actual on-chain execution of the batch.
+
+### Account Abstraction Gas Parameters Override Capability
+
+Identifier:
+
+`accountAbstractionGasParamsOverride`
+
+Interface:
+
+```typescript
+type AAGasParamsOverrideCapabilityParams = Record<
```

Comment:
This is a kind of low-level field override. better separate it to a different ERC.
NonceKey and Paymaster are generic, and don't expose the app directly to version-specific API
(paymaster does require protocol-specific paymaster, but it is encapsulated away from the app)
But this one is very specific. Also, it can't be used without a supporting view APIs to query these values